### ISSUE TITLE
"tokens" key appears twice in data loader and data iterator

### ISSUE THREAD

**Issue by jihunchoi** [allenai/allennlp#3954]
**Describe the bug**
When trying to iterate over a dataset using data loader or data iterator, the "tokens" key appears twice (nested) in each resulting batch.
For example, a batch is expected to be in the following shape:
`{"sentence": {"tokens": torch.tensor(...)}, "label": torch.tensor(...)}`,
however in the current master and v1.0-prerelease version, the shape is:
`{"sentence": {"tokens": {"tokens": torch.tensor(...)}}, "label": torch.tensor(...)}`.

**To Reproduce**
Steps to reproduce the behavior

(When using the data iterator API in v1.0-prerelease)
```
from allennlp.data import dataset_readers, iterators, Vocabulary
reader = dataset_readers.SnliReader()
dataset = reader.read(PATH_TO_DATA_FILE)
vocab = Vocabulary.from_instances(dataset)
iterator = iterators.BasicIterator(batch_size=4)
iterator.index_with(vocab)
loader = iterator(dataset)
batch = next(iter(loader))
# batch is in the following shape:
# {'premise': {'tokens': {'tokens': torch.tensor(...)}}, ...},
```

(When using the data loader API in the master version)
```
from allennlp.data import dataset_readers, dataloader, Vocabulary
reader = dataset_readers.SnliReader()
dataset = reader.read(PATH_TO_DATA_FILE)
vocab = Vocabulary.from_instances(dataset.instances)
dataset.index_with(vocab)
loader = dataloader.DataLoader(dataset, batch_size=4)
batch = next(iter(loader))
# batch is in the following shape:
# {'premise': {'tokens': {'tokens': torch.tensor(...)}}, ...},
```

**Expected behavior**
A batch is expected to be in the following shape:
`{"sentence": {"tokens": torch.tensor(...)}, "label": torch.tensor(...)}`.

**System (please complete the following information):**
 - OS: Ubuntu 18.04
 - Python version: 3.7.6
 - AllenNLP version: master and v1.0-prerelease
 - PyTorch version: 1.4.0

**Comment by jihunchoi** [comment]
I didn't know that the input shape of text field embedder is changed. Never mind!
